modules/telemetry/telemetry_utils.py

```diff
@@ -144,6 +144,9 @@ def __init__(
 
     def run(self):
         while True:
+            # Sleep for 1 ms
+            time.sleep(0.001)
```

Comment:
This section is the part that is the hack fix for now; I'm pretty confident we can come up with a better solution in the near future.

Reply:
Valid; good enough for now! Thanks!